toolkits/google/arcade_google/tools/contacts.py

```diff
@@ -0,0 +1,90 @@
+import asyncio
+from typing import Annotated
+
+from arcade.sdk import ToolContext, tool
+from arcade.sdk.auth import Google
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+
+
+@tool(requires_auth=Google(scopes=["https://www.googleapis.com/auth/contacts.readonly"]))
+async def search_contacts(
+    context: ToolContext,
+    query: Annotated[
+        str,
+        "The search query for filtering contacts.",
+    ],
+    page_size: Annotated[int, "The maximum number of contacts to return"] = 10,
```

Comment:
I see Google caps this at 30. It's a good idea to mention that in the arg annotation. Maybe also highlight that the list of contacts returned might not be exhaustive. Too bad Google won't let you paginate search results... 🤷‍♂️